spec/lib/artifact.spec.js

```diff
@@ -0,0 +1,280 @@
+                resolver(null, {
+                    statusCode: "exampleStatus",
+                    headers: {
+                        location: "exampleLocation"
+                    }
+                });
+            });
+            content = {};
+
+            return instance.uploadFromFile(content).then(() => {
+                expect(fs.createReadStream).not.toHaveBeenCalled();
+                expect(requestOptionsMock.createOptions).toHaveBeenCalled();
+                expect(requestMock.post).toHaveBeenCalled();
+            });
+        });
+        it("calls responseHandler.handleErrorResponse on error", () => {
+            responseHandlerMock.isErrorCode.andReturn(false);
+
+            return instance.uploadFromFile(content).then(() => {
+                expect(responseHandlerMock.handleErrorResponse).toHaveBeenCalled();
+            });
+        });
+        it("calls responseHandler.resolveLink", () => {
+            return instance.uploadFromFile(content).then(() => {
+                expect(responseHandlerMock.resolveLink).toHaveBeenCalled();
+            });
+        });
+    });
+    describe(".download()", () => {
+        it("calls requestOptions.createOptions()", () => {
+            return instance.download().then(() => {
+                expect(requestOptionsMock.createOptions).toHaveBeenCalled();
+            });
+        });
+        it("calls requestPromise.get()", () => {
+            return instance.download().then(() => {
+                expect(requestPromiseMock.get).toHaveBeenCalled();
+            });
+        });
+        it("returns a response body", () => {
+            requestPromiseMock.get.andReturn(bluebird.resolve({
+                body: "responseBody"
+            }));
+
+            return instance.download().then((returnVal) => {
+                expect(returnVal).toBe("responseBody");
+            });
+        });
+        it("calls responseHandler.handleErrorResponse on error", () => {
+            requestPromiseMock.get.andReturn(bluebird.reject());
+
+            return instance.download().then(() => {
+                expect(responseHandlerMock.handleErrorResponse).toHaveBeenCalled();
+            });
+        });
+    });
+    describe(".downloadToFile()", () => {
+        var promise;
+
+        beforeEach(() => {
+            spyOn(fs, "createWriteStream");
+            requestMock.get.andReturn(requestMock);
+            responseHandlerMock.isErrorCode.andReturn(false);
+            responseMock.pipe.andReturn(responseMock);
+        });
+        it("creates a write stream if the file is a string", () => {
+            promise = instance.downloadToFile(downloadLocation).then(() => {
+                expect(fs.createWriteStream).toHaveBeenCalled();
+            });
+            setTimeout(() => {
+                requestMock.emit("response", responseMock);
+                setTimeout(() => {
+                    responseMock.emit("finish");
+                }, 500);
```

Comment:
Only based on precedent. After running these tests probably 100+ times, it seems extremely unlikely it would fail.